### ISSUE TITLE
Standard loss inside custom loss function reports low accuracy metric.

### ISSUE THREAD

**Issue by jubbens** [keras-team/keras#13288]
<em>Please make sure that this is a Bug or a Feature Request and provide all applicable information asked by the template.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.</em>  

**System information**  
- Have I written custom code (as opposed to using example directory):  yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu 18.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  2.0.0rc0
- Keras version:  2.2.4-tf
- Python version:  3.7
- CUDA/cuDNN version:  10.0, 7
- GPU model and memory:  Titan V

You can obtain the TensorFlow version with:  
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"  
You can obtain the Keras version with:  
python -c 'import keras as k; print(k.__version__)'  

**Describe the current behavior**  
Using the Fashion MNIST example from the Tensorflow website.

The following works:

```
model.compile(loss=tf.keras.losses.sparse_categorical_crossentropy, metrics=['accuracy'])
```

If I define the loss like this, it trains and reports similar loss values, but I get random chance accuracy values (~10%):

```
def custom_loss(*args, **kwargs):
    return tf.keras.losses.sparse_categorical_crossentropy(*args, **kwargs)

model.compile(loss=custom_loss, metrics=['accuracy'])
```

**Describe the expected behavior**  

The two cases above should report identical accuracy.

**Code to reproduce the issue**  

As above

**Other info / logs**

**Comment by gowthamkpr** [comment]
@jubbens Can you please share the github gist of this issue. Thanks!

**Comment by jubbens** [comment]
here you go @gowthamkpr, thanks! https://gist.github.com/jubbens/1ce3926fa08b7bf4483e157879f3b816

**Comment by SCP-173-cool** [comment]
@jubbens try replace `metrics=['accuracy']` with `metrics=[kr.metrics.sparse_categorical_accuracy]` where `kr` is `import keras as kr`.
![image](https://user-images.githubusercontent.com/22118890/64675465-d59c6680-d4a5-11e9-9094-ed6c3be1e982.png)
![image](https://user-images.githubusercontent.com/22118890/64675479-db924780-d4a5-11e9-98de-a524878fcec9.png)
